fe/src/main/java/org/apache/doris/planner/OlapScanNode.java

```diff
@@ -447,6 +465,13 @@ private void addScanRangeLocations(Partition partition,
                 scanRangeLocations.addToLocations(scanRangeLocation);
                 paloRange.addToHosts(new TNetworkAddress(ip, port));
                 tabletIsNull = false;
+
+                //for CBO
+                if (replica.getRowCount() != -1) {
+                    cardinality += replica.getRowCount();
+                    totalBytes += replica.getDataSize();
+                }
+                scanBackendIds.add(backend.getId());
```

Comment:
OlapScanNode will put one tablet's all replicas to paloRange, because Coordinator will balance these replicas and try to make that each backend will scan same rows.

However, not all replicas will be read. If table has 3 replica, only one can be read. And this patch will add scan rows three times.

So, you can change cardinality and totalBytes outsize for each tablet other than replica.

Reply:
Yes. I see.  I know only one replica will be read.
the `cardinality ` and `avgRowSize` is only for one replica.  the `cardinality ` and `avgRowSize` will divide replication_num in `computeStats`.

Reply:
I think dividing `FeConstants.default_replication_num` is not a good way. 

Because table's number of replica maybe not equal with it, and even they are equal, when balancing or some BE is down, `tablet.getQueryableReplicas` won't return replica which isn't available. And then, `computeStats` will get a wrong `cardinality`.

Reply:
I agree with you.